ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -880,13 +906,19 @@ def set_waypoints_velocity(self):
                     recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
                                                self.lookahead_wps,
                                                dist_to_tl - (self.dyn_tl_buffer - 1.0))
+                    self.stop_target = self.next_tl_wp
             else:
                 # not sure when this might trigger but not match next if statement - then 
                 # reduce car velocity to 0.0 stoping before lights
                 # if self.state != 'slowdown' and self.state != 'stopped':
                 #    rospy.logwarn("set car state to slowdown at ptr = {}".format(self.final_waypoints_start_ptr))
 
                 if dist_to_tl - self.dyn_tl_buffer > self.min_stop_distance:
+                    if self.state != 'slowdown':
+                        rospy.logwarn("Within stopping distance of TL - Start to slowdown!")
+                        self.stop_target = self.next_tl_wp
+                    else:
+                        rospy.logwarn("we should not get here! {}, {}".format(self.stop_target, self.next_tl_wp))
```

Comment:
This could use a better warning, but I did end up here when the TL_wp was changed to -1 in the middle of a cycle. so it served the purpose.